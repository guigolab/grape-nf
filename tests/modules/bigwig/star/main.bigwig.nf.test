nextflow_process {

    name "Test Process bigwig"
    script "modules/bigwig/star/main.nf"
    process "bw"

    tag "module"

    test("Should convert STAR signal files to BigWig") {

        setup {
            run("fastaIndex") {
                script = "modules/fastaIndex/samtools/main.nf"
                process {
                    """
                    input[0] = "${baseDir}/data/genome.fa.gz"
                    """
                }
            }
            run("signal") {
                script = "modules/bigwig/star/main.nf"
                process {
                    """
                    input[0] = fastaIndex.out
                    input[1] = Channel.from([
                        [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toGenome.bam"), "bam", "GenomeAlignments", true, "MATE2_SENSE"]
                    ])
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = fastaIndex.out
                input[1] = signal.out
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
