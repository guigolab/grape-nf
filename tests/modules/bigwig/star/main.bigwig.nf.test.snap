{
    "Should convert STAR signal files to bigwig": {
        "content": [
            {
                "0": [
                    [
                        "sample3",
                        "test3",
                        [
                            "sample3.Unique.minusRaw.bw:md5,1363d4e278c9024433c2901996ab730a",
                            "sample3.Unique.plusRaw.bw:md5,1b9adcfb8312cde4f51d48d5b0772d01",
                            "sample3.UniqueMultiple.minusRaw.bw:md5,2deb1fd284c866ff5faec6ceba351d56",
                            "sample3.UniqueMultiple.plusRaw.bw:md5,f60e3c597e32fce7b5caf13843080bd6"
                        ],
                        "bigWig",
                        [
                            "MinusRawSignal",
                            "PlusRawSignal",
                            "MultipleMinusRawSignal",
                            "MultiplePlusRawSignal"
                        ],
                        true,
                        "MATE2_SENSE"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-25T12:34:53.753279"
    }
}
